### PR TITLE
fixI(inline-select): Updates invalid text alignment

### LINKED ISSUE #921: Select: inline form validation text alignment
## Detailed description

The positioning of the form validation text on inline `Select`s differs depending on the length of the form label.

> Is this issue related to a specific component? 

`Select`

> What did you expect to happen? What happened instead? What would you like to see changed?

The form validation text should remain consistent

> What version of the Carbon Design System are you using? 

9.1.1

[CodePen example](https://codepen.io/anon/pen/WyJBLM)

I am thinking of right-aligning the validation text for inline invalid `Select`s but would like some input from the core team before implementing a fix for this

### PR DESCRIPTION
**Pull request by alisonjoseph**
## Overview

Resolves #921

Ref #926 

### Added

**Note** This does not solve the issue for IE11, however, I don't see any reason why this fix can't be merged in and we can tackle the IE issue separately. IE11 does not support `auto` when using `display: grid;`

### CODE REVIEW

File: src/components/select/_select.scss [hunk 1]
```diff
@@ -121,14 +121,22 @@
   }
 
   .#{$prefix}--select--inline {
-    display: flex;
-    flex-direction: row;
-    align-items: center;
+    display: grid;
+    grid-template-columns: auto auto;
+    display: -ms-grid;
```
Comment (asudoh) [inline src/components/select/_select.scss]: Did you see autoprefixer doesn't take care of generating `-ms-grid`?

File: src/components/select/_select.scss [hunk 2]
```diff
@@ -121,14 +121,22 @@
...
+      display: flex;
+      flex-direction: row;
+      align-items: center;
+    }
 
     .#{$prefix}--label {
       white-space: nowrap;
       margin: 0 $spacing-xs 0 0;
       font-weight: 400;
+      align-self: center;
```
Comment (asudoh) [inline src/components/select/_select.scss:138]: Wanted to see if Safari is OK with the change here.

**Review by asudoh — COMMENTED** [review]
Great attempt to improve this area @alisonjoseph! Just a couple of comments.

File: src/components/select/_select.scss (hunk 1 above)
Comment (alisonjoseph) [reply src/components/select/_select.scss]: Line 126 actually shouldn't be there, I'll remove. IE11s version of grid doesn't support auto so we can't use it for this purpose. I have IE specific code on line 128 using flexbox that displays the select correctly, with the error message all the way left aligned.

File: src/components/select/_select.scss (hunk 2 above)
Comment (alisonjoseph) [reply src/components/select/_select.scss:138]: Just double checked and this is displaying correctly in Safari.

**Comment by alisonjoseph** [comment]
@emyarod Take a look at my solution here using grid. The fallback for IE is to have the error text all the way left aligned.

**Comment by emyarod** [comment]
Thanks for the heads up @alisonjoseph. I think if that fallback method is alright then grid is probably the way to go here. I synced my branch with your feature branch so there should be no conflicts merging to master. Your changes will fix the validation text alignment issue and I've committed the focus outline fix beforehand

File: src/components/select/_select.scss [hunk 3]
```diff
@@ -121,14 +121,21 @@
   }
 
   .#{$prefix}--select--inline {
-    display: flex;
-    flex-direction: row;
-    align-items: center;
+    display: grid;
+    grid-template-columns: auto auto;
```
Comment (emyarod) [inline src/components/select/_select.scss:125]: Are there any design guidelines that state what the maximum length of the label and/or input field should be (or some kind of ratio between label length and input field length)? If so the `minmax` function may be useful here

Comment (alisonjoseph) [reply src/components/select/_select.scss:125]: I don't believe so, the recommendation for inline select is to be used when there are multiple in a row. I don't believe having the label to the left is a standard use-case that is recommended. http://www.carbondesignsystem.com/components/select/usage

**Comment by alisonjoseph** [comment]
closing this since updates were merged in to #926

**Pull request closed without merging**